### DISCUSSION TITLE
How to configure more than one autoexec?

### DISCUSSION THREAD

**Discussion by patja** [sle118/squeezelite-esp32#177] [Q&A] [answered]
I see references in the logs implying there is a capability to support more than one autoexec command.

I would like to try running a second autoexec command to run the i2cset command to put the ES8388 DAC on an ESP32 A1S board into mono mode.

How can one specify a second autoexec command?  Can the autoexec setting field on the web UI accept an array or series of commands, or is there some other mechanism for specifying a second autoexec command?

**Comment by sle118** [comment, marked as answer]
@patja here is how to do this

- activate the NVS editor (there's a toggle button in the credits tab)
- go to the end of the NVS list, you'll see an empty row
- add a new line for each command you would like to run at start up. Don't skip a number in the sequence since this will signal the system that there are no more entries

**Reply by patja** [reply]
This is great, thanks very much!  I figured it had to be something obvious like this.

This does work for my purpose of setting the ES8388 DAC register 29 to a value of 32 so the DAC operates in mono mode.

But I found that it takes some time for I2C to be available.  Putting i2cset or i2cget commands in autoexec2 and autoexec3 fail with errors such as "i2c driver not installed" and "i2c Read failed".  But if I keep putting more autoexecs in there over and over, eventually they stop failing after the DAC is initialized and it works.  I think it doesn't work until autoexec4 or autoexec5

I attached the relevant section of my log.

[autoexec_i2c_log.txt](https://github.com/sle118/squeezelite-esp32/files/9668666/autoexec_i2c_log.txt)
